SetupDataPkg/ConfDataSettingProvider/ConfDataSettingProvider.c

```diff
@@ -294,6 +294,179 @@ DFCI_SETTING_PROVIDER  mSettingsProvider = {
   (DFCI_SETTING_PROVIDER_SET_DEFAULT)ConfDataSetDefault
 };
 
+/**
+  Set new runtime value to variable storage.
+
+  @param This      Provider Setting
+  @param Value     a pointer to the variable list
+  @param ValueSize Size of the data for this setting.
+  @param Flags     Informational Flags passed to the SET and/or Returned as a result of the set
+
+  @retval EFI_SUCCESS If setting could be set.  Check flags for other info (reset required, etc)
+  @retval Error       Setting not set.
+**/
+EFI_STATUS
+EFIAPI
+RuntimeDataSet (
+  IN  CONST DFCI_SETTING_PROVIDER  *This,
+  IN        UINTN                  ValueSize,
+  IN  CONST UINT8                  *Value,
+  OUT DFCI_SETTING_FLAGS           *Flags
+  )
+{
+  EFI_STATUS  Status = EFI_SUCCESS;
+  CHAR16 *VarName = NULL;
+  CHAR16 *name;
+  EFI_GUID *Guid;
+  UINT32 Attributes;
+  CHAR8 *Data;
+  UINT32 CRC32;
+  UINT32 LenToCRC32;  
+  UINT32 CalcCRC32 = 0;
+  
+  if ((This == NULL) || (This->Id == NULL) || (Flags == NULL) || (Value == NULL)) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  *Flags = 0;
+
+  if (0 != AsciiStrnCmp (This->Id, DFCI_OEM_SETTING_ID__RUNTIME, DFCI_MAX_ID_LEN)) {
+    DEBUG ((DEBUG_ERROR, "RuntimeDataSet was called with incorrect Provider Id (%a)\n", This->Id));
+    return EFI_UNSUPPORTED;
+  }
+
+  for (int i = 0; i < ValueSize;) {

```

Comment:
We do not normally declare new variables in the head of a for loop.

Reply:
Or would a while loop suit better in this case?

Reply:
Yep, a while loop will do fine here, I'll update it.